homeassistant/components/vera/common.py

```diff
@@ -0,0 +1,122 @@
+"""Common vera code."""
+import logging
+from typing import Callable, DefaultDict, List, NamedTuple, Optional, Set, Union
+
+import pyvera as pv
+
+from homeassistant.components.scene import DOMAIN as SCENE_DOMAIN, Scene
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+
+from .const import CONF_CONTROLLER, CONTROLLER_DATAS, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ControllerData = NamedTuple(
```

Comment:
This is part of support for multiple controllers.

Reply:
That's in place to reduce duplicate code that accesses specific indexes in the hass.data dictionary. This named tuple guarantees strict data structures at runtime thus avoiding attempts to access data in None types.